src/modules/app/VirtualMouseModule.js

```diff
@@ -21,11 +21,14 @@ export class VirtualMouseModule extends Events {
     this.patchEvents = patchEvents;
   }
 
-  update(e) {
+  update(e, customX, customY) {
     const rect = this.canvas.getBoundingClientRect();
 
-    this.mouse.x = ((e.clientX - rect.left) / (rect.right - rect.left)) * 2 - 1;
-    this.mouse.y = -((e.clientY - rect.top) / (rect.bottom - rect.top)) * 2 + 1;
+    const x = customX || e.clientX;
+    const y = customY || e.clientY;
+
+    this.mouse.x = ((x - rect.left) / (rect.right - rect.left)) * 2 - 1;
```

Comment:
wait actually, the `* 2 - 1` might just be to set the bounds to `[-1, 1]` or something like that